InjectionIII/AppDelegate.swift

```diff
@@ -0,0 +1,218 @@
+        statusItem.highlightMode = true
+        statusItem.menu = statusMenu
+        statusItem.isEnabled = true
+        statusItem.title = ""
+
+        enabledTDDItem.state = UserDefaults.standard.bool(forKey:UserDefaultsTDDEnabled)
+            ? NSControl.StateValue.on
+            : NSControl.StateValue.off
+        enableVaccineItem.state = UserDefaults.standard.bool(forKey:UserDefaultsVaccineEnabled)
+            ? NSControl.StateValue.on
+            : NSControl.StateValue.off
+
+        setMenuIcon("InjectionIdle")
+        DDHotKeyCenter.shared()?
+            .registerHotKey(withKeyCode: UInt16(kVK_ANSI_Equal),
+               modifierFlags: NSEvent.ModifierFlags.control.rawValue,
+               target:self, action:#selector(autoInject(_:)), object:nil)
+    }
+
+    @IBAction func openProject(_ sender: Any) {
+        _ = application(NSApp, openFile:"")
+    }
+
+    @IBAction func addProject(_ sender: Any) {
+        let open = NSOpenPanel()
+        open.prompt = NSLocalizedString("Add Project Directory", tableName: "Project Directory", comment: "Project Directory")
+        open.canChooseDirectories = true
+        open.canChooseFiles = true
+        if open.runModal().rawValue == NSFileHandlingPanelOKButton {
+            let directory = open.url!.path
+            appDelegate.watchedDirectories.insert(directory)
+            self.lastConnection?.watchDirectory(directory)
+        }
+    }
+
+    @IBAction func toggleTDD(_ sender: NSMenuItem) {
+        toggleState(sender)
+        let newSetting = sender.state == NSControl.StateValue.on
+        UserDefaults.standard.set(newSetting, forKey:UserDefaultsTDDEnabled)
+    }
+
+    @IBAction func toggleVaccine(_ sender: NSMenuItem) {
+        toggleState(sender)
+        let newSetting = sender.state == NSControl.StateValue.on
+        UserDefaults.standard.set(newSetting, forKey:UserDefaultsVaccineEnabled)
+        self.lastConnection?.sendCommand(.vaccineSettingChanged, with:vaccineConfiguration())
+    }
+
+    @IBAction func traceApp(_ sender: NSMenuItem) {
+        toggleState(sender)
+        self.lastConnection?.sendCommand(sender.state == NSControl.StateValue.on ?
+            .trace : .untrace, with: nil)
+    }
+
+    func vaccineConfiguration() -> String {
+        let vaccineSetting = UserDefaults.standard.bool(forKey: UserDefaultsVaccineEnabled)
+        let dictionary = [UserDefaultsVaccineEnabled: vaccineSetting]
+        let jsonData = try! JSONSerialization
+            .data(withJSONObject: dictionary, options:[])
+        let configuration = String(data: jsonData, encoding: .utf8)!
+        return configuration
+    }
+
+    func application(_ theApplication: NSApplication, openFile filename: String) -> Bool {
+        let open = NSOpenPanel()
+        open.prompt = NSLocalizedString("Select Project Directory", tableName: "Project Directory", comment: "Project Directory")
+        //    open.allowsMultipleSelection = TRUE;
```

Comment:
Should this be commend out?

Reply:
Altered the code to accept multiple directories

Reply:
This is true, perhaps we could just remove this line then?

Reply:
removed